src/MultiRegion/cubed_sphere_boundary_conditions.jl

```diff
@@ -2,25 +2,7 @@ using Oceananigans.MultiRegion: number_of_regions
 
 import Oceananigans.BoundaryConditions: fill_halo_regions!
 
-function find_neighboring_panels(grid::ConformalCubedSphereGrid, region)
-    number_of_regions(grid) !== 6 && error("requires cubed sphere grids with 1 region per panel")
-
-    if isodd(region)
-        region_E = mod(region + 0, 6) + 1
-        region_N = mod(region + 1, 6) + 1
-        region_W = mod(region + 3, 6) + 1
-        region_S = mod(region + 4, 6) + 1
-    elseif iseven(region)
-        region_E = mod(region + 1, 6) + 1
-        region_N = mod(region + 0, 6) + 1
-        region_W = mod(region + 4, 6) + 1
-        region_S = mod(region + 3, 6) + 1
-    end
-
-    return (; region_E, region_N, region_W, region_S)
-end
-
-function fill_halo_regions!(field::CubedSphereField{<:Center, <:Center})
+function fill_halo_regions!(field::CubedSphereField{<:Center, <:Center}; multilayer = true)
```

Comment:
what is a `multilayer`?
let's find a way to dispatch

Reply:
introducing a kwarg boolean and an if-else statement within a method affects performance

Reply:
I am sure there is a better way to handle this e.g. dispatch. I use the multilayer kwarg to distinguish between arrays that include a third dimension, like velocities, from those that do not, such as cubed sphere grid metrics.

Reply:
why don't you use their type?